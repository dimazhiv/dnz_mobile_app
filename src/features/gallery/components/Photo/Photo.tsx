import React from 'react';
import { Image, StyleSheet } from 'react-native';

export type PhotoProps = {
  uri: string;
};

function Photo({ uri }: PhotoProps) {
  return <Image source={{ uri: uri }} style={styles.image} />;
}

const styles = StyleSheet.create({
  image: {
    width: 300,
    height: 300
  }
});

export default Photo;
