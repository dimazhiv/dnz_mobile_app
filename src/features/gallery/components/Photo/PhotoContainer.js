import { connect } from 'react-redux';
import { getCurrentPhoto } from '../../selectors';
import Photo from './Photo';

function mapStateToProps(state) {
  return {
    uri: getCurrentPhoto(state)
  };
}

export default connect(mapStateToProps, null)(Photo);
