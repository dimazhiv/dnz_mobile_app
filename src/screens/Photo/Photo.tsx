import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import PhotoComponent from '../../features/gallery/components/Photo/PhotoContainer';

function Photo() {
  return (
    <SafeAreaView style={styles.container}>
      <PhotoComponent />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  }
});
export default Photo;
