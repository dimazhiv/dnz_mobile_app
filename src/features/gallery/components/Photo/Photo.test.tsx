import React from 'react';
import { shallow } from 'enzyme';
import Photo from './Photo';

describe('Snapshot testing Photo component', () => {
  test('correct component view', () => {
    const tree = shallow(<Photo uri={'https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg'} />);
    expect(tree).toMatchSnapshot();
  });
});
