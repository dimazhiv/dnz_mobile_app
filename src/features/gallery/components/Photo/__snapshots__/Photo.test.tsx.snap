// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Photo component correct component view 1`] = `
<Image
  source={
    {
      "uri": "https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg",
    }
  }
  style={
    {
      "height": 300,
      "width": 300,
    }
  }
/>
`;
